### PR TITLE
Issue/119 File Downloads

### PR DESCRIPTION
**Pull request by DanBrink91**
Putting this up so we can start a discussion on how to mock a file download and if we need anything else from this method.

Solves issue #119

### CODE REVIEW

**Comment by DanBrink91** [comment]
Updated with tests

File: canvasapi/file.py [hunk 1]
```diff
@@ -25,3 +25,29 @@ def delete(self):
...
+        """
+        response = self._requester.request(
+            'GET',
+            _url=self.url
+        )
+        return response.text
+
+    def download(self, location):
+        """
+        Download the file to specificed location
```
Comment (jessemcbride) [inline canvasapi/file.py]: specified* 

End the docstring with a period too pls

File: canvasapi/file.py [hunk 2]
```diff
@@ -25,3 +25,29 @@ def delete(self):
...
+            'GET',
+            _url=self.url
+        )
+        return response.text
+
+    def download(self, location):
+        """
+        Download the file to specified location.
+
+        :param location: The path to download to.
```
Comment (jessemcbride) [inline canvasapi/file.py]: Include the type's parameter after the `:param location:`.

```
:type location: str
```

File: tests/test_file.py [hunk 3]
```diff
@@ -7,6 +7,8 @@
 from canvasapi.file import File
 from tests import settings
 from tests.util import register_uris
+from tests.util import cleanup_file
```
Comment (jessemcbride) [inline tests/test_file.py]: Flip import order to make it alphabetical.

File: tests/test_file.py [hunk 4]
```diff
@@ -35,3 +37,19 @@ def test_delete_file(self, m):
...
+
+    # download()
+    def test_download_file(self, m):
+        register_uris({'file': ['file_download']}, m)
+
+        self.file.download('canvasapi_file_download_test.txt')
+        self.assertTrue(isfile('canvasapi_file_download_test.txt'))
+        with open('canvasapi_file_download_test.txt') as downloaded_file:
+            self.assertEqual(downloaded_file.read(), '"file contents are here"')
+        cleanup_file('canvasapi_file_download_test.txt')
```
Comment (Thetwam) [inline tests/test_file.py]: It's technically possible (though unlikely) for this to not run if part of the tests fails. That means the file wouldn't be cleaned up and would be left on the test runner's system. To get around this in other places, we've wrapped stuff in a `try`/`finally`.

**Pull request merged**